src/ccnl-core/src/ccnl-relay.c

```diff
@@ -570,8 +570,9 @@ ccnl_content_add2cache(struct ccnl_relay_s *ccnl, struct ccnl_content_s *c)
     DEBUGMSG_CORE(DEBUG, "ccnl_content_add2cache (%d/%d) --> %p = %s [%d]\n",
                   ccnl->contentcnt, ccnl->max_cache_entries,
                   (void*)c, ccnl_prefix_to_str(c->pkt->pfx,s,CCNL_MAX_PREFIX_SIZE), (c->pkt->pfx->chunknum)? *(c->pkt->pfx->chunknum) : -1);
+
     for (cit = ccnl->contents; cit; cit = cit->next) {
-        if (c == cit) {
+        if (ccnl_prefix_cmp(c->pkt->pfx, 0, cit->pkt->pfx, CMP_EXACT) == 0) {
```

Comment:
I know it's used somewhere else and it's probably jt a matter of taste, but for pointers I usually would expect `NULL` instead of `0` to immediately see that the parameter expects a pointer

Reply:
My bad. This was totally unintentional.